packages/openmrs/src/Utils.js

```diff
@@ -21,18 +21,52 @@ export function request(state, method, path, data, params) {
 
   const options = {
     body: data,
-
     headers: {
       ...headers,
       'content-type': 'application/json',
     },
-
     query: params,
-
     parseAs: 'json',
   };
 
   const url = `${instanceUrl}${path}`;
 
-  return commonRequest(method, url, options).then(response => logResponse(response));
+  return autoFetchRequest(method, url, options);
 }
+
+const autoFetchRequest = async (method, url, options) => {
+  let allResponses;
+  let reqQuery = options?.query;
+  let allowPagination = isNaN(reqQuery?.startIndex);
+
+  do {
+    const response = await commonRequest(method, url, {
+      ...options,
+      query: reqQuery ?? {},
+    });
+    logResponse(response);
+
+    allResponses
```

Comment:
This still needs to be made into an if-else statement.

A ternary expression should always be on the right-hand-side of some expression. It shouldn't be a statement by itself.

Reply:
Made some improvements for the requestOptions

Reply:
Ah maybe I wasn't clear - I'm talking about this:
```
if (allResponses) {
  allResponses.body.results.push(...response.body.results)
} else {
  allResponses = response
}
```

Reply:
Thanks i have update the code accordingly